### PR TITLE
Switch to new surf revparser

### LINKED ISSUE #139: Incorporate new surf revparser
After radicle-dev/radicle-surf#61 landed we should remove the current chain of naive checks.

### PR DESCRIPTION
**Pull request by xla**
This avoids brittle chaining and is closer to the underlying git2 mechanisms.

Closes #139

### CODE REVIEW

File: proxy/Cargo.toml [hunk 1]
```diff
@@ -32,14 +32,15 @@ warp = "0.1"
...
+rev = "4cedafaeb46deee4ec11d0e4b779a1a0d124d599"
 
 [dependencies.radicle-registry-client]
 git = "https://github.com/radicle-dev/radicle-registry.git"
 rev = "8689536f7e30d5e4839141658d2357e2cc550755"
 
 [dependencies.radicle-surf]
-version = "0.2"
+git = "https://github.com/radicle-dev/radicle-surf.git"
+rev = "5c04b846cf53763a70bf0c168ce40293bbce0bd6"
```
Comment (FintanH) [inline proxy/Cargo.toml]: This shouldn't be necessary anymore right?

File: proxy/src/coco.rs [hunk 2]
```diff
@@ -158,14 +158,7 @@ pub fn blob(paths: &Paths, id: &str, revision: &str, path: &str) -> Result<Blob,
...
     // Best effort to guess the revision.
-    if browser
-        .branch(surf::git::BranchName::new(revision))
-        .or(browser.commit(surf::git::Sha1::new(revision)))
-        .or(browser.tag(surf::git::TagName::new(revision)))
-        .is_err()
-    {
-        return Err(error::Error::Git(surf::git::error::Error::NotBranch));
-    };
+    browser.revspec(revision)?;
```
Comment (FintanH) [inline proxy/src/coco.rs]: :kissing_closed_eyes::ok_hand:

File: proxy/src/coco.rs [hunk 3]
```diff
@@ -250,9 +243,9 @@ pub fn commit(paths: &Paths, id: &str, sha1: &str) -> Result<Commit, error::Erro
         project::Project::Git(git_project) => git_project.browser()?,
     };
 
-    browser.commit(radicle_surf::git::Sha1::new(sha1))?;
+    browser.commit(radicle_surf::git::Oid::from_str(sha1)?)?;
 
-    let history = browser.get_history();
+    let history = browser.get();
     let commit = history.0.first();
```
Comment (FintanH) [inline proxy/src/coco.rs]: This can just be `history.first()`

File: proxy/src/coco.rs [hunk 4]
```diff
@@ -368,7 +354,7 @@ pub fn tree(paths: &Paths, id: &str, revision: &str, prefix: &str) -> Result<Tre
     entries.sort_by(|a, b| a.info.object_type.cmp(&b.info.object_type));
 
     let last_commit = if path.is_root() {
-        Some(Commit::from(browser.get_history().0.first()))
+        Some(Commit::from(browser.get().0.first()))
```
Comment (FintanH) [inline proxy/src/coco.rs]: Same here, `get().first()`

File: proxy/src/coco.rs [hunk 5]
```diff
@@ -230,7 +223,7 @@ pub fn local_branches(repo_path: &str) -> Result<Vec<Branch>, error::Error> {
     let mut branches = browser
         .list_branches(None)?
         .into_iter()
-        .map(|b| Branch(b.name.name()))
+        .map(|b| Branch(b.name.name().to_string()))
```
Comment (FintanH) [inline proxy/src/coco.rs]: Note to self: this is just bad... I'll make an issue to have a `.name()` function for `Branch`

Comment (xla) [reply proxy/src/coco.rs]: @FintanH Should I create an issue for it?

Comment (FintanH) [reply proxy/src/coco.rs]: I already did :) radicle-dev/radicle-surf#68

**Pull request merged**